test/loader/test_link_neighbor_loader.py

```diff
@@ -189,20 +189,33 @@ def test_temporal_heterogeneous_link_neighbor_loader():
     data['paper'].x = torch.arange(100)
     data['paper'].time = torch.arange(data['paper'].num_nodes)
     data['author'].x = torch.arange(100, 300)
+    data['author'].time = torch.arange(data['author'].num_nodes)
 
     data['paper', 'paper'].edge_index = get_edge_index(100, 100, 500)
     data['paper', 'author'].edge_index = get_edge_index(100, 200, 1000)
     data['author', 'paper'].edge_index = get_edge_index(200, 100, 1000)
 
+    with pytest.raises(ValueError, match=r'`edge_label_time` is specified .*'):
+        loader = LinkNeighborLoader(data, num_neighbors=[-1] * 2,
+                                    edge_label_index=('paper', 'paper'),
+                                    batch_size=32, time_attr='time')
+
+    # With edge_time:
+    edge_time = torch.arange(data['paper', 'paper'].edge_index.size(1))
```

Comment:
I am not sure how easy it is to add, but would be great to have a test that would fail without properly handling of `edge_label_time` and overriding `node_time_dict`.

Reply:
with `num_neighbors>0` this test used to fail before this commit https://github.com/pyg-team/pytorch_geometric/pull/5137/commits/fab08082bda9785223c7d65dc8aa1bcb9c65ea58.

Reply:
Yes, not referring to `num_workers > 0`, but to the previous implementation of how we handled temporal sampling in link prediction (without `edge_label_time`).

Reply:
Ah, I see. Now `LinkNeighborLoader` can't do temporal sampling without `edge_label_time`.

Reply:
Modified line 190 in the test. This checks if the `author` nodes have time stamp smaller than the `edge_label_time` and not necessarily smaller than `paper` nodes.